packages/shared/src/components/tooltips/BaseTooltip.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, ReactElement, Ref } from 'react';
+import Tippy, { TippyProps } from '@tippyjs/react';
+import classNames from 'classnames';
+import styles from './BaseTooltip.module.css';
+import { isTouchDevice } from '../../lib/tooltip';
+import { isTesting } from '../../lib/constants';
+import {
+  BaseTooltipContainer,
+  BaseTooltipContainerProps,
+  TooltipPosition,
+} from './BaseTooltipContainer';
+
+const DEFAULT_DELAY_MS = 400;
+const DEFAULT_OUT_ANIMATION = 200;
+
+export const getShouldLoadTooltip = (): boolean =>
+  !isTouchDevice() && !isTesting;
+
+export interface BaseTooltipProps extends TippyProps {
+  container?: Omit<BaseTooltipContainerProps, 'children'>;
+  placement?: TooltipPosition;
+}
+
+export function BaseTooltip(
+  {
+    render,
+    arrow,
+    placement = 'top',
+    delay = DEFAULT_DELAY_MS,
+    container,
+    children,
+    content,
+    ...props
+  }: BaseTooltipProps,
+  ref?: Ref<Element>,
+): ReactElement {
+  const [unMounting, setUnMounting] = useState(false);
+  const onHide = ({ unmount }) => {
+    setUnMounting(true);
+    setTimeout(() => {
+      setUnMounting(false);
+      unmount();
+    }, DEFAULT_OUT_ANIMATION);
+  };
```

Comment:
Why do we need this part?

Reply:
This is for the animation before we actually close out the tooltip itself. Applying some classes to render the `fade-out`